Add external database support for job statistics storage

Anish-Chanda opened the pull request:
This PR adds support for storing job statistics in an external MariaDB/MySQL database instead of the Slurm database's AdminComment field. This way institutions that dont want an external program to be wirting to slurmdb can still use jobstats.

When enabled via config, all job statistics are written exclusively to the external database with no fallback to Slurm. The jobstats command automatically retrieves data from the configured source.

Configuration is simple - just add the external database connection details to config.py and set enabled=True. Backward compatibility is maintained when the feature is disabled.

jdh4 commented:
Thank you for submitting. You caught us during vacation season so some time will be needed.

plazonic commented:
A couple of adjustments before we can merge this:

- can you please add a bit of documentation, say in a separate .md file (standalone.md or whatever) describing how to add/configure this way of operating. It also needs a SQL schema. Either as part of this documentation or as a separate .sql file.
- slurmctldepilog.sh is changed in these patches so that it disables/breaks adding data into AdminComment. It needs to be conditional. Since this is a bash script I am not sure if it is better to try to detect that configs exist for the db in config file or maybe take the existence of the store_jobstats.py as the sign that one should do that to store the data in the database?

Anish-Chanda commented:
Apologies for the delay in addressing the feedback! I've updated `slurmctldepilog.sh` to make the behavior conditional: it now uses external DB exclusively when configured, and falls back to AdminComment storage only if external DB is not set up. I've also updated the documentation to reflect these changes.

plazonic commented:
Looks good to me - thanks for adding the extra info and cleaning it up. After merging will only change the default config to disabled (it anyway is useless without correct db info).
